### PR TITLE
Update the process to set the Neo4j password

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request updates the Ansible role to:
- Use a different syntax for setting the Neo4j password depending on the version of Neo4j installed
- Use `no_log: true` on the Ansible tasks that set the password

## 💭 Motivation and context ##

- In neo4j 5.0 the syntax for setting the password using the `set-initial-password` command has changed
- The password should not be accidentally leaked if there is an error

This code is cribbed from the following commit by @mcdonnnj: https://github.com/cisagov/ansible-role-bloodhound/pull/43/commits/ded4cb4f90441356c505afd2642681906e361138

## 🧪 Testing ##

All automated tests pass.  I have also built a COOL staging AMI with these changes and verified that it functions as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Very slick 👍

File: src/bloodhound.yml [hunk 1]
```diff
@@ -20,10 +20,48 @@
...
+          ansible.builtin.package_facts:
+
+        - name: Get version of the installed Neo4j package
+          ansible.builtin.set_fact:
+            installed_neo4j_version: >-
+              {{ ansible_facts.packages['neo4j'] |
+              map(attribute='version') | first |
+              replace(':', '!') | replace('"', '') }}
+
+        - name: Set neo4j password on version < 5
```
Comment (mcdonnnj) [inline src/bloodhound.yml]: If we're capitalizing this elsewhere.

```suggestion
        - name: Set Neo4j password on version < 5
```

File: src/bloodhound.yml [hunk 2]
```diff
@@ -20,10 +20,48 @@
...
+          ansible.builtin.command:
+            cmd: >
+              /usr/bin/neo4j-admin
+              set-initial-password {{ password }}
+          changed_when: false
+          no_log: true
+          when:
+            - installed_neo4j_version is version("1!5.0", "<", version_type="pep440")
+
+        - name: Set neo4j password on version >= 5
```
Comment (mcdonnnj) [inline src/bloodhound.yml]: If we're capitalizing this elsewhere.

```suggestion
        - name: Set Neo4j password on version >= 5
```

**Review by mcdonnnj — APPROVED** [review]
Thanks for sussing out working with the epoch version string. Two small suggestions for your consideration but otherwise LGTM ✔

**Pull request merged**